### PR TITLE
Determinism test needs to use bootstrap compiler

### LINKED ISSUE #46684: Consider if compiler server should exit on missing analyzer dependency
Context: https://github.com/dotnet/roslyn/pull/46671

The underlying issue here is that when the server sees missing dependencies loading analyzers it considers that a fatal state and begins tearing down the compiler server. I'm no longer convinced that is the right thing to do.

The history there is that when originally authoring the server we were very concerned about assembly load address space pollution. Given every analyzer loads into the same `AppDomain` on desktop over time they can pollute the assembly load space such that a given compilation in the compiler server would fail due to inability to load an analyzer but it would succeed when run on the command line. 

We were very concerned about this space when we originally wrote the server and when in doubt we leaned towards the paranoid interpretation of the error (when in doubt, shut down). 

Thinking through this now i think missing dependencies should not be a fatal error. If it misses in the server, it's going to miss in the command line. Assembly load space pollution can really only cause incorrect dependencies to load (which we handle) but can't really see a way it would cause them to reasonably not load. 

Possibly the implementation of `ShadowCopyAnalyzerAssemblyLoader` needs to change to better support analyzers. Overall though this work around doesn't really scale to customers. Feel like we need a sustainable solution for this.

### PR DESCRIPTION
**Pull request by jaredpar**
For the determinism test to be effective it should be using the
bootstrap compiler. That was the intent of this job but at some point
the bootstrap got disabled. This fixes that and excludes some binaries
that are now failing determinism.

~Tracking issue for the failed binaries #49482~

This also fixes the issue where we should not shut down the server for missing dependencies. The logic for that is covered in the issue #46684

closes #46684

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure  PTAL

File: eng/test-determinism.ps1 [hunk 1]
```diff
@@ -24,9 +22,23 @@ if ($help) {
 
 # List of binary names that should be skipped because they have a known issue that
 # makes them non-deterministic.  
-$script:skipList = @()
-
+$script:skipList = @(
+  # Added to work around https://github.com/dotnet/roslyn/issues/48417
+  "Microsoft.CodeAnalysis.EditorFeatures2.UnitTests.dll",
```
Comment (Youssef1313) [inline eng/test-determinism.ps1:27]: This one was already fixed.

Comment (jaredpar) [reply eng/test-determinism.ps1:27]: That change only fixed the field emit, it missed the parameters which have the same problem.

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
@AArnott this fixes the performance problem where the compiler server does not work for projects with your analyzers installed.

**Comment by AArnott** [comment]
@sharwell I don't recall the problem you're referring to. Can you job my memory?

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler PTAL. The logic for why we should not restart the server for missing dependencies is covered in  #46684.

**Comment by sharwell** [comment]
@AArnott the context is #46671. We had to include custom build logic because the vs-threading analyzers package was tanking build performance at no fault of its own. After this change, the code fixes assembly in that package will no longer negatively impact build throughput.

**Comment by jaredpar** [comment]
@AArnott 

Not sure if this applies to your analyzers but any analyzer DLL that also packaged a code fixer inside it effectively never used the compiler server. The compiler server is very sensitive about analyzer load errors because it has to deal with real problems around analyzers polluting the assembly load space of the process. That is because the server sees all of the analyzers for all compilations that go through it where as the command line csc just sees the analyzers for the current build. 

Quick example of where this can cause a problem. Consider two projects (UtilA and UtilB) which depend on two versions of analyzer FindTheBugz which is not strong name signed. The compiler server can handle either of them just fine but not both. The first one to build loads their version of FindTheBugz one the second just gets the version the first loaded. The compiler server though detects this problem, redirects the build request to the command line compiler and restarts the server. 

One of the load errors the server would shut down on was missing dependencies. That was written originally under the guise of "better safe than sorry". Over time though it's caused subsantial perf and usability problems. After some discussion decided this is actually an unnecessary restriction and this is lifting it.

File: eng/test-determinism.ps1 [hunk 2]
```diff
@@ -24,14 +22,17 @@ if ($help) {
 
 # List of binary names that should be skipped because they have a known issue that
 # makes them non-deterministic.  
-$script:skipList = @()
-
+$script:skipList = @(
+  # Added to work around https://github.com/dotnet/roslyn/issues/48417
+  "Microsoft.CodeAnalysis.EditorFeatures2.UnitTests.dll"
```
Comment (AlekseyTs) [inline eng/test-determinism.ps1:27]: >"Microsoft.CodeAnalysis.EditorFeatures2.UnitTests.dll" [](start = 2, length = 54)

The fix for parameters got merged.

Comment (jaredpar) [reply eng/test-determinism.ps1:27]: Great. I'll either get this changed here or submit a follow up PR to remove this line.

File: src/Compilers/Server/VBCSCompiler/AnalyzerConsistencyChecker.cs [hunk 3]
```diff
@@ -15,19 +15,12 @@ namespace Microsoft.CodeAnalysis.CompilerServer
 {
     internal static class AnalyzerConsistencyChecker
     {
-        private static readonly ImmutableArray<string> s_defaultIgnorableReferenceNames = ImmutableArray.Create("mscorlib", "System", "Microsoft.CodeAnalysis", "netstandard");
-
         public static bool Check(string baseDirectory, IEnumerable<CommandLineAnalyzerReference> analyzerReferences, IAnalyzerAssemblyLoader loader, IEnumerable<string>? ignorableReferenceNames = null)
```
Comment (AlekseyTs) [inline src/Compilers/Server/VBCSCompiler/AnalyzerConsistencyChecker.cs:18]: >ignorableReferenceNames [](start = 170, length = 23)

Is this parameter still used?

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 2), modulo a couple of comments.

File: src/Compilers/Server/VBCSCompiler/AnalyzerConsistencyChecker.cs (hunk 3 above)
Comment (jaredpar) [reply src/Compilers/Server/VBCSCompiler/AnalyzerConsistencyChecker.cs:18]: No. Thought I had deleted that. Will fix.

**Pull request merged**